config/config.go

```diff
@@ -136,7 +136,15 @@ func (tcf *TerragruntConfigFile) convertToTerragruntConfig(terragruntOptions *op
 	tcf.PreHooks.init(tcf)
 	tcf.PostHooks.init(tcf)
 	err = tcf.RunConditions.init(tcf.options)
-	return &tcf.TerragruntConfig, err
+
+	if tcf.Include != nil {
+		return &tcf.TerragruntConfig, err
+	}
+	// If the newly loaded configuration file is not to be merged, we force the merge
+	// process to ensure that duplicated elements will be properly processed
+	newConfig := &TerragruntConfig{options: tcf.options}
```

Comment:
Since the explanation is here. I think you should reverse the condition. I was wondering what was up with the `Include != nil` part. Easier to read like this: 
```
if tcf.Include == nil {
	// If the newly loaded configuration file is not to be merged, we force the merge
	// process to ensure that duplicated elements will be properly processed
	newConfig := &TerragruntConfig{options: tcf.options}
        newConfig.mergeIncludedConfig(tcf.TerragruntConfig, terragruntOptions)
	return newConfig, err
}

return &tcf.TerragruntConfig, err
```

Reply:
Done!